velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -732,9 +732,14 @@ std::pair<std::string, std::string> HiveDataSink::getWriterFileNames(
         connectorQueryCtx_->planNodeId(),
         makeUuid());
   }
-  const std::string writeFileName = isCommitRequired()
+  std::string writeFileName = isCommitRequired()
       ? fmt::format(".tmp.velox.{}_{}", targetFileName, makeUuid())
       : targetFileName;
+  if (insertTableHandle_->tableStorageFormat() ==
+      dwio::common::FileFormat::PARQUET) {
+    targetFileName += ".parquet";
+    writeFileName += ".parquet";
```

Comment:
Isn't it sufficient to only modify the `writeFileName`?
Also `fmt::format` is the preferred API.
Something like this should allow us to keep the const above.
```
if (insertTableHandle_->tableStorageFormat() ==
      dwio::common::FileFormat::PARQUET) {
  return {targetFileName, fmt::format("{}{}", writeFileName, ".parquet")}
}
```

Reply:
@majetideepak Why not modify `targetFileName ` here? The unit tests in TableWirteTest.cpp with following error.  Can we modify both `targetFileName `and `writeFileName`?

```
../../velox/exec/tests/TableWriteTest.cpp:2450: Failure
Expected equality of these values:
  writeFileName
    Which is: "test_cursor 6763_3_2_063f23c4-896d-4806-8555-3c7e87fc8360.parquet"
  targetFileName
    Which is: "test_cursor 6763_3_2_063f23c4-896d-4806-8555-3c7e87fc8360"

Reply:
From the code, it seemed to me that only the writeFilename is being used. Do you know where the targetFileName gets used? The semantics/uses of both these names needs to be documented.

Reply:
If targetFileName also needs the change, we can simply append to that above as the writeFileName is being derived from that.

Reply:
> From the code, it seemed to me that only the writeFilename is being used. Do you know where the targetFileName gets used? The semantics/uses of both these names needs to be documented in a follow-up PR.

@majetideepak Yes, `targetFileName `is not used except in `TableWriteTest`. The explanation regarding `targetFileName `and `writeFileName `has already been provided [here](https://github.com/facebookincubator/velox/blob/fa6de06c826084ce5d938c71b7450c59204c864d/velox/connectors/hive/HiveDataSink.h#L281).

Reply:
> If targetFileName also needs the change, we can simply append to that above as the writeFileName is being derived from that.

@majetideepak If so, the `writeFileName `may be `.tmp.velox.{targetFileName}_5d9e7bdf-0db6-4b13-b976-0652a75afb32` if `isCommitRequired()` is true [here](https://github.com/facebookincubator/velox/blob/fa6de06c826084ce5d938c71b7450c59204c864d/velox/connectors/hive/HiveDataSink.cpp#L735). This will result in the final `writeFileName `being `.tmp.velox.*******.parquet**_5d9e7bdf-0db6-4b13-b976-0652a75afb32`, which is unexpected.

Reply:
@JkSelf thanks for the pointers. Since `writeFileName` is a temporary name, do we need to change that?

Reply:
> @JkSelf thanks for the pointers. Since `writeFileName` is a temporary name, do we need to change that?

@majetideepak Yes. Gluten only use the  `writeFileName` not `targetFileName `